### ISSUE TITLE
GameResults lack proper information on the game result

### ISSUE THREAD

**Issue by anarchuser** [CAU-Kiel-Tech-Inf/backend#319]
GameResult (and, in relation, PlayerScore) need a rework. Ideas include:
- [ ] Always store both players
  - [ ] Store information (enum?) on whether they won, lost or violated
- [ ] If a rule violation happend: Store a MoveMistake to properly identify the kind of violation that happened

Related to https://github.com/CAU-Kiel-Tech-Inf/gui/issues/7

**Comment by xerus2000** [comment]
@anarchuser is this still relevant?

**Comment by anarchuser** [comment]
Yes. As listed above, GameResults only store the _winner_ of a game. That means you have to request the opponent of that player through the Game, which is tedious. Storing both players and whether they won, lost, violated or drawed makes this information more accessible.
Also, the GUI currently has no way of knowing MoveMistakes that happened and thus can't display them.
